src/engines/html5/media-source/adapters/native-adapter.js

```diff
@@ -704,15 +708,29 @@ export default class NativeAdapter extends BaseMediaSourceAdapter {
           } else if (settings.kind === 'captions' && this._config.enableCEA708Captions) {
             settings.label = settings.label || captionsTextTrackLabels.shift();
             settings.language = settings.language || captionsTextTrackLanguageCodes.shift();
+            settings.available = captionAvailable;
```

Comment:
I missed this in the ABR PR, this `available` attribute looks very funky to me but I understand it serves us here
IMO we shouldn't extend the external object we expose.
I would have kept the internal full track object and expose an external one which is filtered with the only available tracks instead of exposing it to all with this flag.